core/google/cloud/exceptions.py

```diff
@@ -250,3 +259,44 @@ def _walk_subclasses(klass):
     code = getattr(_eklass, 'code', None)
     if code is not None:
         _HTTP_CODE_TO_EXCEPTION[code] = _eklass
+
+_GRPC_ERROR_MAPPING = {
+    StatusCode.UNKNOWN: InternalServerError,
+    StatusCode.INVALID_ARGUMENT: BadRequest,
+    StatusCode.DEADLINE_EXCEEDED: GatewayTimeout,
+    StatusCode.NOT_FOUND: NotFound,
+    StatusCode.ALREADY_EXISTS: Conflict,
+    StatusCode.PERMISSION_DENIED: Forbidden,
+    StatusCode.UNAUTHENTICATED: Unauthorized,
+    StatusCode.RESOURCE_EXHAUSTED: TooManyRequests,
+    StatusCode.FAILED_PRECONDITION: PreconditionFailed,
+    StatusCode.ABORTED: Conflict,
+    StatusCode.OUT_OF_RANGE: BadRequest,
+    StatusCode.UNIMPLEMENTED: MethodNotImplemented,
+    StatusCode.INTERNAL: InternalServerError,
+    StatusCode.UNAVAILABLE: ServiceUnavailable,
+    StatusCode.DATA_LOSS: InternalServerError,
+}
+
+
+@contextlib.contextmanager
+def _catch_remap_gax_error():
```

Comment:
- I meant to say "before", not "for" (sorry, big time DERP there)
- `_GRPC_ERROR_MAPPING` is definitely "end-of-module" but it's used here at call time, not definition time, so that's not an out-of-order issue